### PR TITLE
feat(delete-job-endpoint): Added a delete job endpoint to the Fossolo…

### LINKED ISSUE #1787: Looking for a REST API to kill (or stop) the job
For example, after I triggered decider & reuser job together through REST API, if the reuser agent was failed, the decider agent which depended on reuser would stay at "pending" state forever. Thus, is there an API to force the status for a specific job

### PR DESCRIPTION
**Pull request by Hero2323**
…gy API

It works based on Job ID and Job Queue ID. It also deletes Job Queues that depend ond the Job Queue to be deleted This fixes #1787

## Description

Added a delete endpoint to the fossology API

### Changes

Added a new endpoint `/jobs/{id}/{queue}` that deletes a job queue given the job queue and it's ID. It also deletes all queues depending on this job queue.

Also Added documentation to the `openapi.yaml` file and changed the version number.

Also modified the `index.php` file to add the endpoint to the list of routes.

## How to test

- Schedule a new Job
- Send a request to the localhost/repo/jobs/{id}/{queue} with the job id and job queue you'd like to delete

This fixes #1787 

<a href="https://gitpod.io/#https://github.com/fossology/fossology/pull/2407"></a>

### CODE REVIEW

File: src/www/ui/api/Controllers/JobController.php [hunk 1]
```diff
@@ -167,6 +167,69 @@ public function createJob($request, $response, $args)
...
+
+    foreach ($JobQueue as $job) {
+      if (in_array($queueId, $job["depends"])) {
+       array_push($dependentJobs, $job["jq_pk"]);
+      }
+    }
+
+    /* Delete All jobs in dependentJobs */
+    foreach ($dependentJobs as $job) {
+      $Msg = "\"" . _("Killed by") . " " . $_SESSION[AUTH::USER_NAME] . "\"";
```
Comment (GMishx) [inline src/www/ui/api/Controllers/JobController.php:221]: Please avoid using magic variable `$_SESSION` in API. Use following functions instead to get user object.

$userId = $this->restHelper->getUserId();
$userName = $this->restHelper->getUserDao()->getUserName($userId);

**Review by GMishx — CHANGES_REQUESTED** [review]
Minor change needed.

Please squash two commits into 1 for this PR.

**Comment by Hero2323** [comment]
Added the optimizations, fixed the end of line white spaces and incorrect indentation.

**Review by GMishx — APPROVED** [review]
Changes looks good.

**Review by GMishx — APPROVED** [review]
Tested, working as expected.

**Pull request merged**